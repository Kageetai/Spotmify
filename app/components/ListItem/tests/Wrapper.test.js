import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Wrapper from '../Wrapper';

describe('<Wrapper />', () => {
  it('should match the snapshot', () => {
    const wrapper = renderer.create(<Wrapper />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an <li> tag', () => {
    const wrapper = mount(<Wrapper />);
    expect(wrapper.find('li').type()).toEqual('li');
  });

  it('should have a className attribute', () => {
    const wrapper = mount(<Wrapper />);
    expect(wrapper.find('li').prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = mount(<Wrapper id={id} />);
    expect(wrapper.find('li').prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = mount(<Wrapper attribute="test" />);
    expect(wrapper.find('li').prop('attribute')).toBeUndefined();
  });
});
