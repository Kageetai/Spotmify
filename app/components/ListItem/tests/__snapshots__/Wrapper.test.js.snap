// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should match the snapshot 1`] = `
.c0 {
  width: 100%;
  height: 3em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-top: 1px solid #eee;
}

.c0:first-child {
  border-top: none;
}

<li
  className="c0"
/>
`;
